lib/variable/comparison.cpp

```diff
@@ -18,16 +18,31 @@ namespace scipp::variable {
 
 namespace {
 Variable _values(Variable &&in) { return in.has_variances() ? values(in) : in; }
+
+template <class T, class... Remaining>
+std::optional<Variable>
+try_isclose_spatial(const Variable &a, const Variable &b, const Variable &rtol,
+                    const Variable &atol, const NanComparisons equal_nans) {
+  if (a.dtype() == dtype<T>)
+    return std::optional(
+        all(isclose(a.elements<T>(), b.elements<T>(), rtol, atol, equal_nans),
+            Dim::InternalStructureComponent));
+  if constexpr (sizeof...(Remaining) > 0)
+    return try_isclose_spatial<Remaining...>(a, b, rtol, atol, equal_nans);
```

Comment:
Any reason for the use of recursion, rather than a simple fold-expression?

Reply:
Nope, I just got a bit rusty with templates, it seems

Reply:
Actually, I don't think there is a nice way to do this because `std::optional` cannot easily be chained to return the first one that has a value.

Reply:
We used this elsewhere, but sure it is better: https://github.com/scipp/scipp/blob/847788f7bf835db8b4de06ff35793f0dd86da9f6/lib/variable/include/scipp/variable/variable.h#L197-L202

Reply:
That seems to rely more on the optimizer to remove the extra variable objects. And I personally don't think this is easier to understand than a recursion.